packages/actor-rdf-resolve-hypermedia-links-queue-wrapper-info-occupancy/lib/LinkQueueLogger.ts

```diff
@@ -0,0 +1,209 @@
+        url: link.url,
+        reachability_criteria: reachabilityCriteriaCurrentLink === null ? null : reachabilityCriteriaCurrentLink[0],
+        reachability_criteria_dynamic_info: reachabilityCriteriaCurrentLink === null ?
+          undefined :
+          reachabilityCriteriaCurrentLink[1],
+        timestamp: Date.now(),
+        parent: parentStatisticLink,
+      };
+      this.appendReachabilityStatistic(statisticLink, 'pushEvent');
+      const event: ILinkQueueEvent = {
+        type: EventType.Push,
+        link: statisticLink,
+        query: this.query,
+        queueStatistics: {
+          size: this.getSize(),
+          reachabilityRatio: this.reachabilityRatio,
+        },
+      };
+
+      this.materialize(event);
+    }
+    return resp;
+  }
+
+  public override pop(): ILink | undefined {
+    const link = super.pop();
+    if (link !== undefined) {
+      const reachabilityCriteriaCurrentLink = LinkQueueLogger.getReachability(link);
+      const statisticLink: IURLStatistic = {
+        url: link.url,
+        reachability_criteria: reachabilityCriteriaCurrentLink === null ? null : reachabilityCriteriaCurrentLink[0],
+        reachability_criteria_dynamic_info: reachabilityCriteriaCurrentLink === null ?
+          undefined :
+          reachabilityCriteriaCurrentLink[1],
+        timestamp: Date.now(),
+      };
+      this.appendReachabilityStatistic(statisticLink, 'popEvent');
+      const event: ILinkQueueEvent = {
+        type: EventType.Pop,
+        link: statisticLink,
+        query: this.query,
+        queueStatistics: {
+          size: this.getSize(),
+          reachabilityRatio: this.reachabilityRatio,
+        },
+      };
+
+      this.materialize(event);
+    }
+    return link;
+  }
+
+  /**
+   * Materialize the current event into the logger
+   * @param {ILinkQueueEvent} event - Current event
+   */
+  private materialize(event: ILinkQueueEvent): void {
+    const jsonEvent = { ...event, type: EventType[event.type] };
+    this.logger.trace(LinkQueueLogger.LINK_QUEUE_EVENT_NAME, { data: JSON.stringify(jsonEvent) });
+  }
+}
+
+/**
+ * The type of event
+ */
+export enum EventType {
+  Push,
+  Pop,
+}
+
+/**
+ * A link queue event
+ */
+interface ILinkQueueEvent {
+  type: EventType;
+  link: IURLStatistic;
+  query: Algebra.Operation;
+  queueStatistics: IQueueStatistics;
+}
+/**
+ * Statistic of the link queue
+ */
+interface IQueueStatistics {
+  size: number;
+  reachabilityRatio: IReachabilityRatio;
+}
+/**
+ * Ratio of the reachability criteria of the links in the queue
+ */
+interface IReachabilityRatio {
+  pushEvent: Record<string, number>;
+  popEvent: Record<string, number>;
+}
+/**
+ * Optional parameters necessitating special processing
+ */
+interface IOptionalLinkQueueParameters {
+}
+
+/**
+ * Information about an URL
+ */
+interface IURLStatistic {
+  url: string;
+  reachability_criteria: string | null;
+  reachability_criteria_dynamic_info?: object;
```

Comment:
What is this?

Reply:
It's extra information about the acquisition of the link like for predicate actor there are information about which predicate from the set capture the links. It is not in one object to facilitate the compilation of the ratio of criterion. But it could also be in one object.